atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamTableDefinitionBuilder.java

```diff
@@ -98,13 +101,11 @@ public TableDefinition build() {
             return new TableDefinition() {{
                 javaTableName(streamTableType.getJavaClassName(prefix));
                 rowName();
-                if (hashFirstRowComponent) {
-                    hashFirstRowComponent();
-                }
-                rowComponent("id",            idType);
+                    hashFirstNRowComponents(min(numberOfComponentsHashed, 1));
```

Comment:
This looks like a backcompat break. I think we should have `hashFirstNRowComponents(0)` if nothing is specified, `hashFirstNRowComponents(1)` if `hashFirstRowComponent` is specified, else `hashFirstNRowComponents(numberOfComponentsHashed)`. All of this should be captured by `numberOfComponentsHashed` already, so min looks odd here, as you are disallowing the previously allowed 0 value. 

same for the below instances.

Reply:
Talked offline, this makes sense as these stream tables have only one rowComponent.